lib/src/working_copy.rs

```diff
@@ -511,6 +559,64 @@ impl TreeState {
             })
             .collect();
 
+        let fsmonitor_clock_needs_save = fsmonitor_kind.is_some();
+        let changed_files: Option<Vec<PathBuf>> = match fsmonitor_kind {
+            None => None,
+            Some(FsmonitorKind::Test { changed_files }) => Some(changed_files),
+            #[cfg(feature = "watchman")]
+            Some(FsmonitorKind::Watchman) => match self.query_watchman(self.watchman_clock.clone())
```

Comment:
Let `query_watchman()` read `self.watchman_clock` itself? I.e. those changes from the last commit to this commit.